### PR TITLE
if command is ran with a compose file, apply the compose model

### PR DESCRIPTION
**Pull request by ndeloof**
**What I did**
pass project to start/restart/pause/unpause if user ran command with a compose.yaml file, not just project-name
backend will then apply model (active profiles) and not consider all resources with project label

**Related issue**
see https://github.com/docker/compose/issues/9286

Fixes https://github.com/docker/compose/issues/9705
Fixes https://github.com/docker/compose/issues/9671

**(not mandatory) A picture of a cute animal, if possible in relation with what you did**
image

### CODE REVIEW

**Comment by laurazard** [comment]
I fixed the merge conflicts and added some of the tests I'd already written for the other PR, this one covers the other use case and applies the same logic to other commands 🥳

**Review by milas — APPROVED** [review]
🥳

**Comment by laurazard** [comment]
Ahh! Need to fix some tests 😅

**Comment by laurazard** [comment]
While looking into why this was failing when applying a similar logic to `compose ps` I noticed some inconsistencies regarding how the `-p` flag is handled vs the `COMPOSE_PROJECT_NAME` env var.

The reason why some tests were failing was that we are running `compose ps` with the `COMPOSE_PROJECT_NAME` env var set. Using `projectOptions.toProjectName(...)`, this worked because before attempting to load the project,  it checks for the env var and doesn't attempt to load the project otherwise. However, the introduction of `projectOptions.projectOrName(...)` causes some issues here because it doesn't attempt to check the env var, and instead always loads the project unless the passed `projectOptions` struct already has a name set. The inconsistencies here are due to the fact that the `projectOptions.ProjectName` field is set when running `compose -p [project-name]`, but not when running `COMPOSE_PROJECT_NAME=[project-name] compose`.

I think we should handle these in the same way, probably in the parent `compose` command, and set that value regardless of where the project name comes from. This would also make some other things more consistent, ie `compose -p [name] down` works even if there is no `compose.yaml` in the directory but `COMPOSE_PROJECT_NAME=[name] compose down` fails.

I made some changes to `projectOrName()` in this PR to fallback to the previous behavior so the tests can pass, but I think some of this logic should be cleaned up and moved up for more consistency.

**Comment by ndeloof** [comment]
> compose -p [name] down works even if there is no compose.yaml in the directory but COMPOSE_PROJECT_NAME=[name] compose down fails

Why would you like the latter to fail? IMHO once `COMPOSE_PROJECT_NAME` is set, all compose command should be able to run based on existing resources, same as passing `-p` flag. But maybe I'm missing some corner cases...

**Comment by ndeloof** [comment]
LGTM with @laurazard additions

File: pkg/e2e/fixtures/start-stop/other.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+services:
+  a-different-one:
+    image:  nginx:alpine
+  and-another-one:
+    image:  nginx:alpine
```
Comment (glours) [inline pkg/e2e/fixtures/start-stop/other.yaml:5]: nit: add a new line

**Review by glours — APPROVED** [review]
LGTM 👍

File: pkg/e2e/fixtures/start-stop/other.yaml (hunk 1 above)
Comment (laurazard) [reply pkg/e2e/fixtures/start-stop/other.yaml]: done :)

**Pull request merged**